src/Campaigns/Migrations/MigrateFormsToCampaignForms.php

```diff
@@ -131,24 +131,35 @@ public function createCampaignForForm($formData): void
                 'date_created' => $formCreatedAt,
             ]);
 
-        DB::table('give_campaign_forms')
-            ->insert([
-                'form_id' => $formId,
-                'campaign_id' => $campaignId,
-                'is_default' => true,
-            ]);
+        $this->addCampaignFormRelationship($formId, $campaignId, true);
     }
 
     /**
      * @param $data
      */
     protected function addUpgradedFormToCampaign($data): void
     {
-        DB::table('give_campaign_forms')
-            ->insert([
-                'form_id' => $data->migratedFormId,
-                'campaign_id' => $data->campaignId,
-            ]);
+        $this->addCampaignFormRelationship($data->migratedFormId, $data->campaignId, false);
```

Comment:
At this point, there is only one form per campaign. You don't have to check if a campaign-form relationship exists. I'd suggest rolling back this commit.

Reply:
Without this checking, I getting the errors described here: https://lw.slack.com/archives/C04SLRDD9CK/p1727723671542809

Reply:
So there can be a problem with data integrity. Okay, we can use `distinct` on query when fetching meta table data.  This will be much more performant than always checking if the form exist in table, which is basically an N+1 problem.

Reply:
The `distinct` works fine with the query generated by the `getFormData()` method, but does not for the query generated by the `getUpgradedFormData()` method -  here is a version a bit different (only added the status column) of this last query:

```
SELECT forms.post_status, forms.ID AS formId, campaign_forms.campaign_id AS campaignId, give_formmeta_attach_meta_migratedFormId.meta_value AS migratedFormId 
FROM wp_posts AS forms 
LEFT JOIN wp_give_formmeta give_formmeta_attach_meta_migratedFormId ON ID = give_formmeta_attach_meta_migratedFormId.form_id AND give_formmeta_attach_meta_migratedFormId.meta_key = 'migratedFormId' 
RIGHT JOIN wp_give_campaign_forms campaign_forms ON campaign_forms.form_id = forms.ID 
WHERE forms.post_type = 'give_forms' AND give_formmeta_attach_meta_migratedFormId.meta_value IS NOT NULL
```

As you can see on the attached screenshot, this query is returning the same `migratedFormId` for different forms. So, we are trying to insert this same `migratedFormId` on the DB multiple times and it's throwing the duplicated entry errors that prevent the migration from getting finished.

Honestly, I don't know the steps to replicate the problem and make you get a similar result on your end. But it makes me think that if it's happening on my end, could happen for customers as well.

The problem only happens in forms with the `trash` status, so we could remove those for the query, but I don't think it's a good idea since the form legacy view has the restore option available - check the secound attached screenshot.

For now, I'll just left the distinct that solves the problem partially and will allow us move forward with this PR, but later I'll keep checking this problem to see if there is something we can do here.

Partially resolved in: da3cd569bb101bb564b97ee4e286af18036e5f9b

![image](https://github.com/user-attachments/assets/9c25caef-3c4e-472f-933c-61150dd676f7)

![image](https://github.com/user-attachments/assets/db116e6c-4d37-4735-afe6-a4fc5dcc3c72)

Reply:
Ok, so `getUpgradedFormData` is used to get upgraded form data, right? Then we are missing something really important here, and that's where condition with post status value. 

Currently, this query will return all forms regardless of the status of the form. So the easy solution here would be to add another `WHERE` condition where we want forms only with the status `upgraded`. 
```php
->where('forms.post_status', 'upgraded')
```

Reply:
Oh yeah! It makes much more sense. 😅

Resolved in 9ac7b978746a1b487ffb4ba47c472147d538d6a1

Reply:
Ops, I just noticed the `MigrateFormsToCampaignFormsTest::testUpgradedFormsAreNotDefault` is failing after this change.

Reply:
In fact, the proper where is this one:

`->where('forms.post_status', 'publish')`

Because this query is getting the forms that already are using the Visual Form Builder and the `migratedFormId` meta key value is reference for they source, the legacy form where they came from - this source that have the `upgraded` status.

Reply:
Yeah, makes sense.